src/User.ts

```diff
@@ -84,11 +84,19 @@ export class User {
     }
   }
 
+  public async loadWeb3Account (): Promise<void> {
+    if (this._web3.eth.defaultAccount) {
+      this.address = this._web3.eth.defaultAccount
```

Comment:
the address belongs to the signing service. So address should probably just forward to txSigner.address.

Reply:
Then user does not have a direct dependency of web3.